packages/react-core/src/components/NumberInput/__tests__/NumberInput.test.tsx

```diff
@@ -60,4 +61,90 @@ describe('numberInput', () => {
     );
     expect(asFragment()).toMatchSnapshot();
   });
+
+  test('renders negative value', () => {
+    render(<NumberInput value={-999} />);
+    expect(screen.getByDisplayValue('-999')).toBeInTheDocument();
+  });
+
+  test('renders decimal value', () => {
+    render(<NumberInput value={5.5} />);
+    expect(screen.getByDisplayValue('5.5')).toBeInTheDocument();
+  });
+
+  test('renders negative decimal value', () => {
+    render(<NumberInput value={-5.2} />);
+    expect(screen.getByDisplayValue('-5.2')).toBeInTheDocument();
+  });
+
+  test('renders unit with negative number', () => {
+    const { asFragment } = render(<NumberInput value={-8} unit="%" unitPosition="after" />);
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  test('renders unit & position with negative number', () => {
+    const { asFragment } = render(<NumberInput value={-8} unit="$" unitPosition="before" />);
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  test('renders unit with decimal number', () => {
+    const { asFragment } = render(<NumberInput value={0.64} unit="%" unitPosition="after" />);
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  test('renders unit & position with decimal number', () => {
+    const { asFragment } = render(<NumberInput value={0.64} unit="$" unitPosition="before" />);
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  test('renders unit with negative decimal number', () => {
+    const { asFragment } = render(<NumberInput value={-12.4} unit="%" unitPosition="after" />);
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  test('renders unit & position with negative decimal number', () => {
+    const { asFragment } = render(<NumberInput value={-12.4} unit="$" unitPosition="before" />);
+    expect(asFragment()).toMatchSnapshot();
+  });
```

Comment:
@tlabaj are these tests for all these different possible combinations of input values and units something you were looking for?

I feel like the tests that I've highlighted here could be trimmed down to just a test for `unit` with `"before"` and a test for `unit` with `"after"` since the unit and its position aren't dependent on what the input value is.

Reply:
@wise-king-sullyman I think that is fine to trim that down.  I was not specific enough with my comment.
I was actually looking for test to verify the trimming of the leading zeros works with different values ( decimal, negative numbers) as stated in description.

Reply:
@wise-king-sullyman @tlabaj I'll remove the tests on units and position then and it should be all set for review

Reply:
@andyyvo I think that @tlabaj is looking for tests which verify that the new behavior with leading zeros being removed works as expected, which I'm not currently seeing in the test suite. Please correct me if I'm wrong about my read of your last comment @tlabaj.

To do so I'm guessing you'll need to use `userEvent` to type a leading zero into the input then click outside of the input, then make an assertion about the value of the input, rather than just being able to pass it into `NumberInput` as a value.

Reply:
@wise-king-sullyman that is correct.  This could also be done with a cypress integration test.